packages/compiler/core/program.ts

```diff
@@ -85,7 +95,7 @@ export async function createProgram(
     hasError() {
       return error;
     },
-    onBuild(cb) {
+    onValidate(cb) {
```

Comment:
In this scenario, How do we imagine that existing emitters will implement OnValidate?  It seems like the default will be to   essentially do what is being done in OnBuild - this means that existing emitters will run on-build functionality twice when used in the `--emit` as well as in the spec imports

Reply:
not sure I fully get this. Emitter would need to update `onBuild` to `onEmit` and linter `onBuild` -> `onValidate`.
Or did you mean for older version of the emitters? If so I think they are not meant to be compatible with a later minor version of cadl for now so it won't be possible to use the latest compiler with and older library.